android/java/org/chromium/chrome/browser/crypto_wallet/activities/AssetDetailActivity.java

```diff
@@ -444,13 +444,17 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
     }
 
     private void adjustButtonsVisibilities() {
-        if (Utils.allowBuyAndSwap(mChainId)) {
+        if (Utils.allowBuyOrSwap(mChainId, BuySendSwapActivity.ActivityType.BUY)) {
+            mBtnBuy.setVisibility(View.VISIBLE);
+        } else {
+            mBtnBuy.setVisibility(View.GONE);
+        }
+        if (Utils.allowBuyOrSwap(mChainId, BuySendSwapActivity.ActivityType.SWAP)) {
```

Comment:
nit: Better to create separate methods for buy and swap `IsBuySupported` and `IsSwapSupported` as there is a different use case now for swap

Reply:
I separate `allowBuyOrSwap` into two functions and add a list for `SWAP_SUPPORTED_NETWORKS`.